R/iterate.R

```diff
@@ -0,0 +1,118 @@
+#' Perform requests iteratively, generating new requests from previous responses
+#'
+#' @description
+#' `r lifecycle::badge("experimental")`
+#'
+#' `req_perform_iteratively()` iteratively generates and performs requests,
+#' using a callback function, `next_req`, to define the next request based on
+#' the current request and response. You will probably want to it pair with an
+#' [iteration helper][iterate_with_offset] and use a
+#' [multi-response handler][resps_combine] to process the result.
+#'
+#' @inheritParams req_perform
+#' @param next_req A function that takes the previous response (`resp`) and
+#'   request (`req`) and returns a [request] for the next page or `NULL` if
+#'   the iteration should terminate.
+#' @param max_reqs The maximum number of requests to perform. Use `Inf` to
+#'   perform all requests until `next_req()` returns `NULL`.
+#' @param progress Display a progress bar? Use `TRUE` to turn on a basic
+#'   progress bar, use a string to give it a name, or see [progress_bars] to
+#'   customise it in other ways.
+#' @param path Optionally, path to save body of request. This should be
```

Comment:
```suggestion
#' @param path Optionally, path to save the body of request. This should be
```

(the same applies for `req_perform()`)